champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/gladiator/sword/BlockToss.java

```diff
@@ -0,0 +1,379 @@
+import me.mykindos.betterpvp.champions.champions.skills.types.InteractSkill;
+import me.mykindos.betterpvp.core.combat.events.CustomDamageEvent;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.components.champions.events.PlayerUseSkillEvent;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.gamer.Gamer;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilDamage;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilPlayer;
+import me.mykindos.betterpvp.core.utilities.UtilTime;
+import me.mykindos.betterpvp.core.utilities.model.ProgressBar;
+import me.mykindos.betterpvp.core.utilities.model.display.PermanentComponent;
+import me.mykindos.betterpvp.core.world.blocks.WorldBlockHandler;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.bukkit.entity.*;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.util.Vector;
+import java.util.*;
+
+
+@Singleton
+@BPvPListener
+public class BlockToss extends ChannelSkill implements InteractSkill, CooldownSkill {
+    private final WeakHashMap<Player, Long> gap = new WeakHashMap<>();
+    private final WeakHashMap<FallingBlock, Player> fallingBlocks = new WeakHashMap<>();
+    private final WeakHashMap<FallingBlock, Boolean> activeBlocks = new WeakHashMap<>();
+    private final WeakHashMap<FallingBlock, Set<Entity>> hasHit = new WeakHashMap<>();
+    private final WeakHashMap<Player, BlockToss.BlockData> charging = new WeakHashMap<>();
+    private final WorldBlockHandler blockHandler;
+    private double duration;
+    private double baseCharge;
+    @Getter
+    private double charge;
+    private double baseDamage;
+
+    @Inject
+    public BlockToss(Champions champions, ChampionsManager championsManager, WorldBlockHandler blockHandler) {
+        super(champions, championsManager);
+        this.blockHandler = blockHandler;
+    }
+    @Override
+    public String getName() {
+        return "Block Toss";
+    }
+    public Boolean checkTags(FallingBlock block, String s){
+        for (String curr : block.getScoreboardTags()) {
+            if (curr.equals(s)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "Hold right click with a Sword on a block to channel",
+                "",
+                "Charge a block to be thrown at the enemies dealing damage.",
+                "",
+                "Higher Charge time = Stronger Block and Farther Distance",
```

Comment:
Charge time should also be indicated. IIRC this is not consistent for all skills, those will need a change to.